jwt/audience.go

```diff
@@ -1,79 +1,33 @@
 package jwt
 
 import (
-	"errors"
-	"fmt"
-	"net"
-	"net/url"
-	"strings"
 )
 
-// Audience must be the base URL from the request including protocol, domain,
-// and port if applicable for the domains you specify in your IAP proxy. For
-// example, https://example.com or https://foo.example.com:port.
-type Audience url.URL
+// From the IAP docs at https://cloud.google.com/iap/docs/signed-headers-howto:
+// Audience must be a string with the following values:
+// * App Engine: /projects/PROJECT_NUMBER/apps/PROJECT_ID
+// * Compute Engine and Container Engine: /projects/PROJECT_NUMBER/global/backendServices/SERVICE_ID
+type Audience string
 
-// NewAudience returns an Audience from a URL.
-func NewAudience(u *url.URL) *Audience {
+// NewAudience returns an Audience from a string.
+func NewAudience(u *string) *Audience {
 	return (*Audience)(u)
 }
 
 // Sanitize normalizes the structure of the Audience's URL and validates it.
 func (aud *Audience) Sanitize() error {
-	aud.Scheme = strings.ToLower(aud.Scheme)
```

Comment:
Since we are being backwards incompatible, maybe we don't need this whole function anymore and just use `Validate` directly?